File: CodingSample/DotNet/Fundamentals/Runtime/ReflectionTest/BankFinLib/Finance/Loans.cs
namespace Finance;

public static class Loans
{
	public static decimal GetMonthlyInstallment(decimal loan, int duration, float rate)
	{
		double i = rate / 1200;
		int m = 12 * duration;
		double g = i / (1 - Math.Pow(1 + i, -m));
		return (decimal)g * loan;
	}
}

File: CodingSample/Java/Core/Runtime/Reflection/bankfin/src/main/java/finance/Loans.java
package finance;

public class Loans {
    
    public static double monthlyInstallment(double loan, int period, float rate) {
        float i = rate / 1200;
        int m = 12 * period;
        return loan * i / (1 - Math.pow(1 + i, -m));
    }
}
